### PR TITLE
adapt for large datasets

### PR DESCRIPTION
**Pull request by griembauer**
`v.divide.training_validation` failed when using large vector maps as input (>~20000 points):
`OSError: [Errno 7] Argument list too long: 'v.extract`
This is now checked and handled chunk-wise. 
PR created together with @AnikaBettge

### CODE REVIEW

File: v.divide.training_validation.py [hunk 1]
```diff
@@ -71,7 +101,7 @@ def main():
         'v.db.select', map=input,
         column=column, flags='c')
 
-    grass.message("Select point for each class...")
+    grass.message("Select points for each class...")
```
Comment (neteler) [inline v.divide.training_validation.py:104]:
```suggestion
    grass.message("Selecting points for each class...")
```

File: v.divide.training_validation.py [hunk 2]
```diff
@@ -88,15 +118,15 @@ def main():
...
-    grass.run_command(
-        'v.extract', input=input, cats=','.join(training_cats), output=training)
-    grass.run_command(
-        'v.extract', input=input, cats=','.join(validation_cats), output=validation)
+    grass.message(_("Extracting training points..."))
+    extract_data(input, training, training_cats, 1)
+    grass.message(_("Extracting validation points..."))
+    extract_data(input, validation, validation_cats, 2)
 
     grass.message("Divided data in <%s> and <%s>" % (training, validation))
```
Comment (neteler) [inline v.divide.training_validation.py:126]:
```suggestion
    grass.message("Divided data into <%s> and <%s>" % (training, validation))
```

**Comment by neteler** [comment]
Please add yourself to the list of authors in https://github.com/mundialis/v.divide.training_validation/blob/62e77252055b0edecd37032fc3cabe768d5eb9d2/v.divide.training_validation.html#L14
:-)

File: v.divide.training_validation.py [hunk 3]
```diff
@@ -45,20 +45,50 @@
...
+    if len(cats) > 20000:
+        newcol = 'train_val_%s' % (os.getpid())
+        columns_existing = grass.vector_columns(options['input']).keys()
+        if newcol not in columns_existing:
+            grass.run_command(
+                'v.db.addcolumn', map=input, columns='%s INTEGER' % (newcol))
+        n = 500
+        for i in range(0, len(cats), n):
+            cats_list = cats[i:i+n]
+            where_list = ["cat='%s'" % (cat) for cat in cats_list]
```
Comment (metzm) [inline v.divide.training_validation.py:75]: The cat column is integer, thus not `cat='%s'` but `cat=%s`.

File: v.divide.training_validation.py [hunk 4]
```diff
@@ -45,20 +45,50 @@
...
+            grass.run_command(
+                    'v.db.update', where=' OR '.join(where_list),
```
Comment (metzm) [inline v.divide.training_validation.py:77]: Instead of `OR` you can use something like `where cat in ((,).join(catlist))`, this is much shorter.

Comment (neteler) [reply v.divide.training_validation.py:77]: Thanks!

Comment (neteler) [reply v.divide.training_validation.py:77]: addressed in #3

**Pull request merged**